src/Microsoft.Azure.SignalR.Common/ServiceCommunicationScope.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Azure.SignalR.Common
+{
+    /// <summary>
+    /// Represents a generic nestable scope for grouping operations with the same service communication policy.
+    /// Derived classes must ensure that the service protocol can understand and transmit this policy to the service.
```

Comment:
> Derived classes must ensure that the service protocol can understand and transmit this policy to the service.

What does this mean?

Reply:
Agree, the wording is confusing/misleading - its not the derived class' responsibility to ensure that the service protocol can understand... What I wanted to say was: if for example `class TraceScope : ServiceCommunicationScope<bool> {..}` is designed to carry a flag signifying whether we should tell the service to trace messages within this scope, then whatever part that uses the service protocol to send messages to the service should check for this scope's flag and transmit this information to the service... I am afraid even this explanation is not clear enough until we start using it in tracing.